### DISCUSSION TITLE
S100103 - nothing on sceen/no power on. Square Shape

### DISCUSSION THREAD

**Discussion by kgreene1983** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#253] [Q&A]
Hi 

So I have a S100103 (Helsinki)

I'm using the PCB, I've added the 10k resistor. 

I used the stock compile and in Model.h its pre-2021 (the instruction guide just says yes in the "Supported or #define model most likely to work"

I've add it to the unit I can connect to the module, but the screen on the unit does nothing. 

On the software the temperature reads 0, and if i try to toggle the Unit F/C it wont set... 

What have I done wrong?

**Comment by visualapproach** [comment]
Hi. Try the MIAMI2021

**Reply by kgreene1983** [reply]
Tried from the Miami2021, but still nothing on the screen at all. 

The toggle buttons now works, but nothing happens on the unit. 

I've checked my solder joints... Nothing. 

Seen other people on here using it with this model and no problems.

Could it be TXS0108E? Being faulty, stupidly I soldered direct to the PCB.

**Reply by kgreene1983** [reply]
Any advice from anyone. No screen/buttons is a sign of software or hardware error.

**Reply by visualapproach** [reply]
Did you add three resistors as mentioned in the build instructions? Put them in series on pin 3,4,5 to the dsp. (The clock, cs and data pins)

**Reply by kgreene1983** [reply]
ok... So add those 560-ohm resistors.. 

So now on power up I get the music playing. The screen and button is still dead. Tried pre-2021 and MIAMI2021 updated it OTA.

So I am getting closer, any other ideas :(

***** update electronics store gave me 560k Resistors... Will see what happens when I fix that. 

Is there a way to check which model has loaded?

**Comment by kgreene1983** [comment]
It works.

Confirming it needs the 560 Ohm resistors, could that be added to the instructions for this model?

 Thanks for everyone's help. 

Two Questions..

Can I see what model i uploded?

Can I view wifi strength?

**Reply by visualapproach** [reply]
Great! Did you use the txs0108 level shifter or the transistor version?
You can't tell what model you uploaded other than knowing what you uploaded and if it's working or not.
You can see Wi-Fi strength here:
![image](https://user-images.githubusercontent.com/22987250/167284678-ec0633f3-14eb-4a1b-b5ba-9ae0d08f2903.jpeg)

**Comment by kgreene1983** [comment]
I used the txs0108.

With the extra resistor I covered them with heat shrink as the legs where
across the H4 pins.

If you ever do a re-design of your great PCB would be good to have some
holes for the resistors.. People who don't need it can just solder a jumper
wire.

Thanks for your amazing software

**Reply by visualapproach** [reply]
Yes, thanks for the suggestion.

**Comment by ktiedt** [comment]
Damn, looks like I have the same problem. I totally missed the 560 ohm resistor notes too :( 

However, I put together a 2nd PCB because my first one was trash (only soldering iron i could find was a soldering gun that broke after 30mins... Didn't even get the website to load on it after plugging it into the pump... but new soldering iron came in today and was able to put 2nd board together rather quickly and much more cleanly... it powers on and works as described here, no pump controls lighting up or anything soooo guess I'm ordering resistors, I have literally above and below 560 but no combinations that can make 560 or close to it :(

**Reply by paulknox** [reply]
Hi there... I have successful build the board for a 6 wire Cancun Airjet Spa. I added the 10K resistor and the board boots as expected.

When I plug inline nothing happens so I'm wondering if I need to add a 560 as well.

I'm just not sure where the resistor should go so if anyone has a picture for reference that would be amazing!

I used a TXS0108E - thanks for a great project!

Paul

**Reply by visualapproach** [reply]
Hi, 
"Nothing happens" is kind of a vague description, but
have you tried to `#define` the other models first? In that case this is where to put the resistors.
Header H2:
1----------DSP (edit: I meant to write DSP and not CIO here)
2----------DSP
3-resistor-DSP
4-resistor-DSP
5-resistor-DSP
6----------DSP

/Visualapproach

**Reply by kgreene1983** [reply]
I'm glad I'm not the only one.

@visualapproach when you get time it would be good to add to your instructions that pin out information ie what each pin is and which to add the resistor.

Thanks again for all your help, the project is great.

I highly recommend ordering the PCB, it was my fist time ordering one and it was super easy

**Reply by paulknox** [reply]
Sorry for the vagueness - so when the board is plugged in via usb power it connects to my wifi and I can load all pages. When I plug it in to the actual unit the unit display shows nothing and the board has no lights and I cannot obviously connect. The unit display unit is not responsive - when I power up the unit I can he the solenoid in the unit. When I take the board out it works as expected.

I have the Miami selected in the code but its a Cancun unit

**Reply by ktiedt** [reply]
I can say, that my "does nothing" was slightly different, I can speak from 2 boards experience here... my first board (really terrible solder job with a bulky solder gun) would power on, (blinky blue led) but I never saw it connect to wifi, the 2nd board which was soldered with a much more proper iron and done nice and clean... would connect to wifi and report details over the web page but no indicators on the pump would light up or function.

It did take what seemed like a minute or 2 for the 2nd board to popup and serve the webpage, but it did without any errors (beyond just not functioning when you click things like heater).

My 560 ohm resistors are arriving today, and I left my pump unscrewed so I can quickly swap it back in and test... I'll have an update later this evening (6-7hrs or so from now)

**Reply by visualapproach** [reply]
@kgreene1983 wrote
> I'm glad I'm not the only one.

I understand your feeling but I disagree. It would be nice if you were the only one with trouble ;-)

> @visualapproach when you get time it would be good to add to your instructions that pin out information ie what each pin is and which to add the resistor.

It's on my todo list which is longer than my list of free time for the moment.
> Thanks again for all your help, the project is great.

Yay, you are very welcome

**Reply by visualapproach** [reply]
@paulknox thank you for the clarification. Since you have a rebranded model it's hard to know exactly what's the problem. Does it have a model number somewhere? But it should boot up and connect to wifi anyway. Check with a DMM that +5V and GND is in the right place. Also check for shorts between pins. 

@ktiedt fingers crossed

**Reply by ktiedt** [reply]
oh yeah and I believe ours has a different model number even... I need to confirm that today too, but what I wrote down early on was s100105 (but its definitely a [Miami](https://www.amazon.com/gp/product/B00HRT863U))

**Reply by ktiedt** [reply]
Just plugged back in the board with the 3 resistors and I got the same behavior... website comes online, no critical errors, but control panel on pump did nothing, and pump did not respond to anything in the app. I didnt have time to do much troubleshooting at that point, so I'll try a few model changes tomorrow when I can leave the board in and the pump not heating, for longer.

**Reply by visualapproach** [reply]
@ktiedt Maybe you have one of those devices that only work with the other level converter (and Eric's PCB or a DIY protoboard)?

**Reply by ktiedt** [reply]
I hope not... that sounds like even more weeks of delays :( I am hoping to have time to do more in depth troubleshooting with it later today.

And I did confirm, the correct model # previously written down, is not currently listed in any discussions here... so yay my luck!

<img width="427" alt="Screen Shot 2022-05-11 at 12 49 00 PM" src="https://user-images.githubusercontent.com/97242/167914080-ba17a78a-b9b4-412d-bf3b-fa5891893f92.png">

**Reply by ktiedt** [reply]
Well my troubleshooting got cut short by work again, and because my OTA updates are not working... seems to be a config error but literally followed the PDF on that respect. Will have to do more testing in that regard I guess.

**Reply by paulknox** [reply]
Update

Resistors arrived and I went to install and noticed that 2 connector wires had broken so I removed and re soldered. At this point I checked all my connectors with a DMM and found the my DuPont ground was faulty. Replaced and I can confirm that the board now works!

My only outstanding issue is working out the port forwarding. I assumed that the board is working on port 80 but when I setup all the port forwarding and try and connect either outside or inside using my DNS server and port the page doesn't load fully and doesn't display a temp. If I connect from within my network with the ip it works fine??? Any ideas??

![image](https://user-images.githubusercontent.com/836850/171721172-302f527b-b29c-4109-8338-d10a19a9e335.jpeg)

**Reply by visualapproach** [reply]
Glad to hear that it works!
from main.h:
```
/** a websocket object that listens on port 81 */
WebSocketsServer webSocket(81);
```
Meaning you would need to forward both 80 and 81. But I don't recommend it from a security standpoint. Try a VPN instead.
Maybe you're already aware of that but to all other reading this, it's an important note!

Cheers

**Reply by paulknox** [reply]
Little update regarding the port forwarding... I have port forwarding setup on the broadband modem which points to my mesh router. I have the mesh router port forwarding to the IP of the module but when I try and connect it looks like the MQTT never finishes loading.

![image](https://user-images.githubusercontent.com/836850/172565655-8fc584e8-4fdf-40aa-8a1d-96c6b725baaa.jpeg)

When I check on a web browser the websocket.js:41 is showing as pending and nothing happens. When I put my external url in with port 8081 which is forwarded to internal 81 I see the 'This is a Websocket server only!' message...

![image](https://user-images.githubusercontent.com/836850/172566720-51cbef7f-83be-4200-b8fe-180f66470c03.jpeg)

Any ideas??

**Reply by ktiedt** [reply]
> When I check on a web browser the websocket.js:41 is showing as pending and nothing happens. When I put my external url in with port 8081 which is forwarded to internal 81 I see the 'This is a Websocket server only!' message...
> 
> Any ideas??

Did you forward both the websocket port and the http port? If you want everything to work from outside your home wifi, then you need to open that port up too otherwise your browser wont be able to establish that connection for the real time data that the websocket is used for.
